### PR TITLE
Add policy distillation

### PR DESCRIPTION
**Pull request by minseop4898**
Modified distillation buffer for larger capacity

### CODE REVIEW

File: configs/pong_no_frameskip_v4/dqn.py [hunk 1]
```diff
@@ -57,7 +57,7 @@
                 kappa=1.0,
                 output_activation=identity,
                 # NoisyNet
-                use_noisy_net=True,
+                use_noisy_net=False,
```
Comment (MrSyee) [inline configs/pong_no_frameskip_v4/dqn.py:60]: should restore prev config

File: rl_algorithms/common/buffer/distillation_buffer.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+import numpy as np
+import torch
+from torch.utils.data import DataLoader, Dataset
+
+device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+
+class DistillationBuffer:
+    """Fixed-size buffer to store experience tuples.
```
Comment (MrSyee) [inline rl_algorithms/common/buffer/distillation_buffer.py:16]: should add document in detail.

File: load_buffer.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import pickle
```
Comment (MrSyee) [inline load_buffer.py:1]: should remove this file

File: rl_algorithms/common/buffer/distillation_buffer.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+"""Distillation buffer."""
```
Comment (MrSyee) [inline rl_algorithms/common/buffer/distillation_buffer.py:2]: should add detail documentation and reference.
e.g.
https://github.com/medipixel/rl_algorithms/blob/4c6cf558712a940998aca6068c8fb4eaa6b5cd03/rl_algorithms/dqn/agent.py#L1

File: rl_algorithms/distillation/dqn_agent.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+"""DQN agent for episodic tasks in OpenAI Gym.
```
Comment (MrSyee) [inline rl_algorithms/distillation/dqn_agent.py:2]: should modify detail documentation and reference.

File: rl_algorithms/common/buffer/distillation_buffer.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+    def sample_for_diltillation(self):
+        """Sample a batch of state and Q-value for student's learning."""
+        assert self.buffer_size >= self.batch_size
+
+        return next(self.dataloader)
+
+
+class DistillationDataset(Dataset):
+    """pytorch Dataset class for random batch data sampling.
```
Comment (MrSyee) [inline rl_algorithms/common/buffer/distillation_buffer.py:76]: should capitalize the first letter

File: run_pong_no_frameskip_v4.py [hunk 7]
```diff
@@ -23,9 +23,12 @@ def parse_args() -> argparse.Namespace:
     parser.add_argument(
         "--cfg-path",
         type=str,
-        default="./configs/pong_no_frameskip_v4/dqn.py",
+        default="./configs/pong_no_frameskip_v4/distillation_dqn.py",
```
Comment (MrSyee) [inline run_pong_no_frameskip_v4.py:26]: restore prev path

File: run_pong_no_frameskip_v4.py [hunk 8]
```diff
@@ -23,9 +23,12 @@ def parse_args() -> argparse.Namespace:
...
         help="config path",
     )
+    parser.add_argument(
```
Comment (MrSyee) [inline run_pong_no_frameskip_v4.py:29]: I think Lunarlander also need buffer path flag.
Or distillation configs and option of lunarlander should be removed.

File: rl_algorithms/distillation/dqn_agent.py [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+    """DQN for policy distillation."""
+
+    # pylint: disable=attribute-defined-outside-init
+    def _initialize(self):
+        """Initialize non-common things."""
+        self.softmax_tau = 0.01
+        self.learner = build_learner(self.learner_cfg)
+
+        self.buffer_path = (
+            f"./distillation_buffer/{self.log_cfg.env_name}/"
```
Comment (MrSyee) [inline rl_algorithms/distillation/dqn_agent.py:42]: How about to change path?

`./distillation_buffer -> ./data/distillation_buffer`

File: .gitignore [hunk 10]
```diff
@@ -10,4 +10,8 @@ MUJOCO_LOG.TXT
 # build
 build
 dist
-*.egg-info
\ No newline at end of file
+*.egg-info
+
+# data
+data/distillation_buffer*.pkl
```
Comment (MrSyee) [inline .gitignore:16]: According to this comment https://github.com/medipixel/rl_algorithms/pull/259/files#r452582414,
should remove useless ignore.

File: Makefile [hunk 11]
```diff
@@ -1,14 +1,14 @@
...
-	env PYTHONPATH=. pytest --pylint --flake8 --cov=tests --ignore=checkpoint --ignore=distillation_buffer --ignore=wandb --ignore tests/integration
+	env PYTHONPATH=. pytest --pylint --flake8 --cov=tests --ignore=checkpoint --ignore=data --ignore=wandb --ignore tests/integration
 
 integration-test:
 	env PYTHONPATH=. pytest tests/integration --cov=tests
 
 format:
 	black . --exclude checkpoint wandb
-	isort -y --skip checkpoint --skip wandb --skip distillation_buffer
+	isort -y --skip checkpoint --skip wandb --skip data
```
Comment (minseop4898) [inline Makefile:11]: should remove `--skip data`

File: run_lunarlander_v2.py [hunk 12]
```diff
@@ -76,9 +79,7 @@ def parse_args() -> argparse.Namespace:
     parser.add_argument(
         "--save-period", type=int, default=100, help="save model period"
     )
-    parser.add_argument(
-        "--episode-num", type=int, default=1500, help="total episode num"
-    )
+    parser.add_argument("--episode-num", type=int, default=15, help="total episode num")
```
Comment (minseop4898) [inline run_lunarlander_v2.py:82]: should restore

File: run_lunarlander_v2.py [hunk 13]
```diff
@@ -28,6 +28,9 @@ def parse_args() -> argparse.Namespace:
         default="./configs/lunarlander_v2/dqn.py",
         help="config path",
     )
+    parser.add_argument(
+        "--buffer-path", type=str, default=None, help="buffer path",
```
Comment (minseop4898) [inline run_lunarlander_v2.py:32]: should change name and help description

**Pull request merged**